theano/gpuarray/tests/test_dnn.py

```diff
@@ -2298,3 +2298,233 @@ class Cudnn_grouped_conv(Grouped_conv_noOptim):
     conv2d_gradi_op = dnn.GpuDnnConvGradI
     flip_filter = False
     is_dnn = True
+
+
+def test_dnn_spatialtf():
+    if not dnn.dnn_available(test_ctx_name):
+        raise SkipTest(dnn.dnn_available.msg)
+
+    utt.seed_rng()
+
+    """
+    Spatial Transformer implementation using Theano from Lasagne
+    Original author: skaae (https://github.com/skaae)
+    """
+    def spatialtf_cpu(theta, inp, scale_height, scale_width, border_mode='nearest'):
```

Comment:
It seems you have finally implemented a CPU version for spatial transformation ! Could you move it to CPU part of theano (somewhere relevant in `theano.tensor`) ? Maybe a right place should be `theano/tensor/spatialtf.py`, as this could be used even for non-NNET operations.

If so, add a consistent docstring (infos about args) and a doc entry. We will discuss about it later, but this seems close to be a third part for your GSoC !

PS: I have not yet reviewed the whole code.